lib/cacheWrapper.js

```diff
@@ -79,31 +115,33 @@ cacheWrapper._stash = function( deferred, options ) {
 * @param {Object} options - Including segment, key and value.
 * @returns {undefined} - nothing
 */
-cacheWrapper._queueRequest = function( deferred, options ) {
+cacheWrapper._queueRequest = function(deferred, options) {
   // add the request to the queue
-  queryQueue.push( [ deferred, options ] );
-
-  if ( !startingClient ) {
+  queryQueue.push([ deferred, options ]);
+  if (!startingClient) {
     startingClient = true;
-    cacheClient.start( function( error ) {
-      _.forEach( queryQueue, function( queueItem ) {
+    cacheClient.start(function(error) {
```

Comment:
We should handle the cacheClient.start error case here and reject the deferred object, instead of in the subsequent forEach.